File: Java/src/main/java/com/xurxodev/stringcalculator/NumberValidatorByValidateRules.java
package com.xurxodev.stringcalculator;

import com.xurxodev.stringcalculator.exception.ValidatorException;

import java.util.List;

public class NumberValidatorByValidateRules implements StringCalculator.NumberValidator{

    List<ValidateRule> validateRules;

    public NumberValidatorByValidateRules(List<ValidateRule> validateRules){
        this.validateRules = validateRules;
    }

    @Override
    public void validate(List<Integer> numbers) throws ValidatorException {
        for (Integer number:numbers) {
            validate(number);
        }
    }

    private boolean validate(Integer number) throws ValidatorException {
        for (ValidateRule validateRule : validateRules) {
            validateRule.validate(number);
        }

        return false;
    }

    public interface ValidateRule {
        void validate(Integer number) throws ValidatorException;
    }
}

File: Swift/stringcalculator/NumberValidatorByValidateRules.swift
//
//  NumberValidatorByValidateRules.swift
//  stringcalculator
//
//  Created by Jorge Sánchez on 13/10/16.
//  Copyright © 2016 xurxodev. All rights reserved.
//

import Foundation

class NumberValidatorByValidateRules:NumberValidator {
    
    let validateRules:[ValidateRule]
    
    init (validateRules:[ValidateRule]){
        self.validateRules = validateRules
    }
    
    func validate(numbers:[Int]) throws{
        
        for number in numbers {
            try validate(number)
        }
    }
    
    private func validate(number:Int) throws{
        
        for validateRule in validateRules {
            try validateRule.validate(number)
        }
    }
}